### PR TITLE
Update Guide_Py_Performance.ipynb

### PR DESCRIPTION
**Pull request by frqqyy**
- Corrected some spelling mistakes
- Added a note to guide the installation of the "memory_profiler" module, otherwise there might be an error

### CODE REVIEW

**Comment by frqqyy** [comment]
I'm sorry that I changed two files on the same branch accidentally...

File: doc/guide/Guide_PythonDos-n-Donts.ipynb [hunk 1]
```diff
@@ -407,10 +407,10 @@
     }
    },
    "source": [
-    "A detailed instruction can be found [here](http://docs.spyder-ide.org/current/pylint.html).<br>\n",
+    "A detailed instruction can be found [here](https://pylint.org/).<br>\n",
```
Comment (emanuel-schmid) [inline doc/guide/Guide_PythonDos-n-Donts.ipynb:410]:
```suggestion
    "A detailed instruction can be found [here](https://docs.spyder-ide.org/current/panes/pylint.html).<br>\n",
```
https://pylint.org doesn't work for me.

File: doc/guide/Guide_PythonDos-n-Donts.ipynb [hunk 2]
```diff
@@ -447,7 +447,7 @@
     "<br><br>\n",
     "In general, there should be no errors and warnings left, and the overall code quality should be in the \"green\" range (somewhere above 5 or so).\n",
     "\n",
-    "There are [advanced options](http://docs.spyder-ide.org/current/pylint.html#advanced-options) to configure the type of warnings and other settings in pylint."
+    "There are advanced options to configure the type of warnings and other settings in pylint."
```
Comment (emanuel-schmid) [inline doc/guide/Guide_PythonDos-n-Donts.ipynb:450]:
```suggestion
    "There are [advanced options](https://docs.spyder-ide.org/current/panes/pylint.html#advanced-options) to configure the type of warnings and other settings in pylint."
```

Comment (frqqyy) [reply doc/guide/Guide_PythonDos-n-Donts.ipynb:450]: Thanks a lot!

**Pull request merged**